transformations/transformations/pool_allocator.py

```diff
@@ -493,13 +512,18 @@ def _create_stack_allocation(self, stack_ptr, stack_end, ptr_var, arr, stack_siz
 
         # Build expression for array size in bytes
         dim = arr.dimensions[0]
+        if isinstance(dim, RangeIndex):
+            dim = Sum((dim.upper, Product((-1, dim.lower)), 1))
```

Comment:
codecov suggests that case is currently not covered in the tests. It could be reasonably straightforward to add this to the new test by making one of the kernel arguments have a `2:klev+1` dimension for example? It shouldn't change the overall analysis result but would ensure this behaviour does not get broken in the future.

Reply:
Ah, this one is a bit puzzling: the dimension tested is strictly the first one (horizontal) and thus never included in the stack (I think - please shout if I'm wrong on this). As such it seems redundant and I opted to remove that offending check instead.